src/osd/PG.cc

```diff
@@ -4322,6 +4349,12 @@ void PG::scrub_compare_maps()
 
   // ok, do the pg-type specific scrubbing
   _scrub(authmap, missing_digest);
+  if (!scrubber.store->empty()) {
+    dout(10) << __func__ << ": updating scrub object" << dendl;
+    ObjectStore::Transaction t;
+    scrubber.store->flush(&t);
+    osd->store->apply_transaction(osr.get(), std::move(t));
```

Comment:
i am not sure how to remove the "in-memory" scrub results from the MapCacher. so i choose to wait. and i think if we choose to add the scrub object to the temp objects, it will be cleaned up by the ReplicatedPG::on_change(). so even i guess if the object to which the omap is attached get removed, the omap attached to it will also be killed. so we will be fine even we have transactions yet completed when we are entering a new pg interval?